mne/decoding/time_gen.py

```diff
@@ -914,18 +544,681 @@ def _predict(X, estimators):
+                              color=color, xlabel=xlabel, ylabel=ylabel,
+                              legend=legend, chance=chance, label=label)
+
+
+class TimeDecoding(_GeneralizationAcrossTime):
+    """Train and test a series of classifiers at each time point to obtain a
+    score across time.
+
+    Parameters
+    ----------
+    picks : array-like of int | None
+        The channels indices to include. If None the data
+        channels in info, except bad channels, are used.
+    cv : int | object
+        If an integer is passed, it is the number of folds.
+        Specific cross-validation objects can be passed, see
+        sklearn.cross_validation module for the list of possible objects.
+        Defaults to 5.
+    clf : object | None
+        An estimator compliant with the scikit-learn API (fit & predict).
+        If None the classifier will be a standard pipeline including
+        StandardScaler and a linear SVM with default parameters.
+    times : dict | None
+        A dictionary to configure the training times:
+
+            ``slices`` : np.ndarray, shape (n_clfs,)
+                Array of time slices (in indices) used for each classifier.
+                If not given, computed from 'start', 'stop', 'length', 'step'.
+            ``start`` : float
+                Time at which to start decoding (in seconds). By default,
+                min(epochs.times).
+            ``stop`` : float
+                Maximal time at which to stop decoding (in seconds). By
+                default, max(times).
+            ``step`` : float
+                Duration separating the start of subsequent classifiers (in
+                seconds). By default, equals one time sample.
+            ``length`` : float
+                Duration of each classifier (in seconds). By default, equals
+                one time sample.
+        If None, empty dict.
+    predict_mode : {'cross-validation', 'mean-prediction'}
+        Indicates how predictions are achieved with regards to the cross-
+        validation procedure:
+            ``cross-validation`` : estimates a single prediction per sample
+                based on the unique independent classifier fitted in the
+                cross-validation.
+            ``mean-prediction`` : estimates k predictions per sample, based on
+                each of the k-fold cross-validation classifiers, and average
+                these predictions into a single estimate per sample.
+
+        Default: 'cross-validation'
+    scorer : object | None
+        scikit-learn Scorer instance. If None, set to accuracy_score.
+    n_jobs : int
+        Number of jobs to run in parallel. Defaults to 1.
+
+    Attributes
+    ----------
+    picks_ : array-like of int | None
+        The channels indices to include.
+    ch_names : list, shape (n_channels,)
+        Names of the channels used for training.
+    y_train_ : np.ndarray, shape (n_samples,)
```

Comment:
just `ndarray`. @kingjr try to be consistent with the rest of the codebase whenever possible